src/mpi/coll/barrier.c

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#include "mpiimpl.h"
+
+
+/* -- Begin Profiling Symbol Block for routine MPI_Barrier */
+#if defined(HAVE_PRAGMA_WEAK)
+#pragma weak MPI_Barrier = PMPI_Barrier
+#elif defined(HAVE_PRAGMA_HP_SEC_DEF)
+#pragma _HP_SECONDARY_DEF PMPI_Barrier  MPI_Barrier
+#elif defined(HAVE_PRAGMA_CRI_DUP)
+#pragma _CRI duplicate MPI_Barrier as PMPI_Barrier
+#elif defined(HAVE_WEAK_ATTRIBUTE)
+int MPI_Barrier(MPI_Comm comm) __attribute__ ((weak, alias("PMPI_Barrier")));
+#endif
+/* -- End Profiling Symbol Block */
+
+/* Define MPICH_MPI_FROM_PMPI if weak symbols are not supported to build
+   the MPI routines */
+#ifndef MPICH_MPI_FROM_PMPI
+#undef MPI_Barrier
+#define MPI_Barrier PMPI_Barrier
+#endif
+
+
```

Comment:
Is this series of newlines intentional? Even if the original code was like this, do we want to keep consecutive newlines even after reconstruction?

Reply:
No, not intentional. It is a left-over from script executions. I'll make a note and update it later. On the hand, the plan next is remove these boiler-plates and generate them by script, so the whitespace issues may not be relevant soon.

Reply:
Okay, then I will ignore.